pkg/apis/datadoghq/v1alpha1/datadogagentdeployment_types.go

```diff
@@ -124,6 +121,11 @@ type DatadogAgentDeploymentSpecAgentSpec struct {
 	// See https://docs.datadoghq.com/agent/guide/agent-configuration-files/?tab=agentv6 for more details.
 	// +optional
 	Checksd *ConfigDirSpec `json:"checksd,omitempty"`
+
+	// Allow to put custom configuration for the agent, corresponding to the datadog.yaml config file
+	// See https://docs.datadoghq.com/agent/guide/agent-configuration-files/?tab=agentv6 for more details.
+	// +optional
+	CustomConfig *string `json:"customConfig,omitempty"`
```

Comment:
It might be worth considering `Agent` (`DatadogAgentDeploymentSpecAgentSpec`) as the place to put this new configuration. Quite possibly the `Confd` and `Checksd` fields should be there as well. @clamoriniere what do you think?

Reply:
All of them are already in `DatadogAgentDeploymentSpecAgentSpec`, or am I missing something?

Reply:
Never mind, you are right I got confused with the diff here. :) One wish from the usage stand point - do you think it is possible to define this in such a way that we just have one field called `Config`? It just feels a bit odd to have both`Config` and then `CustomConfig`, thoughts?

Reply:
IMHO I would not put `CustomConfig` inside `Config`, but we can find another name that may be more explicit. Maybe `DatadogYaml` or `CustomAgentYaml`?

Reply:
is it mandatory to have a pointer here? 

```suggestion
	CustomConfig string `json:"customConfig,omitempty"`
```

Reply:
Yeah we can simplify, it was to aligne with others, but we also already have an optional string without pointer, so I'll update that too.

Reply:
Fixed